ARCHITECTURE.md

```diff
@@ -0,0 +1,103 @@
+
+**balena-sound.input**
+`balena-sound.input` acts as an input audio multiplexer/mixer. It's the default sink on balenaSound, so all plugins that send audio to the `audio` block will send it to this sink by default. This allows us to route audio internally without worring where it came from: any audio generated by a plugin will pass through the `balena-sound.input` sink, so controlling where it sends its audio we are effectively controlling all plugins at the same time. 
+
+**balena-sound.output**
+`balena-sound.output` on the other hand is the output audio multiplexer/mixer. This one is pretty useful in scenarios where there are multiple soundcards available (onboard, DAC, USB, etc). `balena-sound.output` is always wired to whatever the desired soundcard sink is. So even if we dynamically change the output selection, sending audio to `balena-sound.output` will always result in audio going to the current selection. Again, this is useful to route audio internally without worrying about user selection at runtime.
+
+### Standalone
+![](https://raw.githubusercontent.com/balenalabs/balena-sound/master/docs/images/arch-standalone.png)
+
+Standalone mode is easy to understand. You just pipe ` balena-sound.input` to `balena-sound.output` and that's it. Audio coming in from any plugin will find it's way to the selected output. If this was the only mode, we could simplify the setup and use a single sink. Having the two layers however is important for the next mode which is more complicated.
+
+
+### Multiroom
+![](https://raw.githubusercontent.com/balenalabs/balena-sound/master/docs/images/arch-multiroom.png)
+
+Multiroom feature relies on `snapcast` to broadcast the audio to multiple devices. Snapcast has two binaries working alonside, server and client.
+
+Snapcast server expects audio to be written into a FIFO file, so we create an additional sink (`snapcast` sink) that routes audio from `balena-sound.input` into said FIFO file. The server will then read the file and use TCP packets to broadcast audio to all clients that are connected to it, wether they run in the same device or others. Note that when writting into the FIFO file the audio is "exiting" the `audio` block and no longer under PulseAudio's control.
+
+Snapcast client receives the audio from the server and sends it back into the `audio` block, in particular to `balena-sound.output` sink which will in turn send the audio to whatever output was selected by the user.
+
+This setup allows us to decouple the multiroom feature from the `audio` block while retaining it's advantages.
+
+## Plugins
+
+As described above, plugins are the services generating the audio to be streamed/played. Plugins are responsible for sending the audio into the `audio` block, particularily into `balena-sound.input` sink. There are two alternatives for how this can be acomplished. A detailed explanation can be found [here](https://github.com/balenablocks/audio#usage), in our case:
```

Comment:
```suggestion
As described above, plugins are the services generating the audio to be streamed/played. Plugins are responsible for sending the audio into the `audio` block, particularly into `balena-sound.input` sink. There are two alternatives for how this can be accomplished. A detailed explanation can be found [here](https://github.com/balenablocks/audio#usage), in our case:
```